aten/src/ATen/native/cuda/SegmentReduce.cu

```diff
@@ -53,6 +87,11 @@ Tensor _segment_reduce_cuda_kernel(
   auto offsets = _get_complete_sum(lengths);
   auto* offsets_data_ptr = offsets.data_ptr<int64_t>();
 
+  constexpr int threads_per_block = 256;
+  int num_blocks = segment_count / threads_per_block;
```

Comment:
this will not correctly compute num_blocks if `segment_count` is not evenly divisible by `threads_per_block`, so you'd be needlessly doing grid-stride loop. It's better to do `int64_t num_blocks = (segment_count + threads_per_block - 1)/threads_per_block`, and then you can remove grid-stride loop. You can have an assert that segment_count is less than max_blocks * 256, but in reality it'll never fire, max_blocks is 2**31.